src/Shared/DiagnosticIds/DiagnosticIds.cs

```diff
@@ -117,6 +117,11 @@ internal static class AuditReports
         internal const string AUDREPGEN000 = nameof(AUDREPGEN000);
         internal const string AUDREPGEN001 = nameof(AUDREPGEN001);
     }
+
+    internal static class Obsoletions
+    {
+        internal const string ResourceMonitoring = "EXTOBS0001";
```

Comment:
If we have more obsoletions in ResourceMonitoring, what would this mean for this const?
```suggestion
        internal const string EXTOBS0001 = nameof(EXTOBS0001);
```

I suggest doing similar to how the runtime does it - i.e., define the code and the message together. E.g.:
```suggestion
        internal const string IResourceUtilizationPublisherDiagId = nameof(EXTOBS0001);
        internal const string IResourceUtilizationPublisherMessage = "IResourceUtilizationPublisher  API is obsolete and will be removed in a future version. Consider using observable instruments.";
```

Related - do we have any docs on how to use "observable instruments"?

Reply:
Update the link in the readme to point to Resource monitoring metrics https://learn.microsoft.com/dotnet/core/diagnostics/built-in-metrics-diagnostics#microsoftextensionsdiagnosticsresourcemonitoring, which I will update separately in the dotnet/docs repo

Reply:
Hope it all makes sense now